clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/UpdateFetchPositionsEvent.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.internals.events;
+
+import org.apache.kafka.clients.consumer.internals.SubscriptionState;
+
+/**
+ * Event to update the position to fetch from. This will use the committed offsets if available.
+ * If no committed offsets exist, it will use the partition offsets.
+ *
+ * <p/>
+ *
+ * The event completes with a boolean value indicating if all assigned partitions already had
+ * valid fetch positions (based on {@link SubscriptionState#hasAllFetchPositions()}).
+ */
+public class UpdateFetchPositionsEvent extends CompletableApplicationEvent<Boolean> {
+
+    /**
+     * Deadline to complete the UpdateFetchPositionsEvent. If the event does not complete before
+     * this time, it will be completed exceptionally with a TimeoutException.
+     */
+    private final long deadlineMs;
```

Comment:
The `CompletableApplicationEvent` already has a `deadlineMs` to control the event's lifetime. What's the benefit to reproducing that logic in this subclass?

Reply:
Not needed indeed, this was unintentionally leftover, removed.